MediaGalleryUi/view/adminhtml/web/js/grid/columns/image/insertImageAction.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+/* global FORM_KEY, tinyMceEditors */
+define([
+    'jquery',
+    'wysiwygAdapter'
+], function ($, wysiwyg) {
+    'use strict';
+
+    return {
+
+        /**
+         * Insert provided image in wysiwyg if enabled, or widget
+         *
+         * @param {Object} record
+         * @param {Object} config
+         * @returns {Boolean}
+         */
+        insertImage: function (record, config) {
+            var targetElement;
+
+            if (record === null) {
+                return false;
+            }
+            targetElement = this.getTargetElement(config.targetElementId);
+
+            if (!targetElement.length) {
+                window.MediabrowserUtility.closeDialog();
+                throw 'Target element not found for content update';
+            }
+
+            if (targetElement.is('textarea')) {
+                $.ajax({
+                    url: config.onInsertUrl,
+                    data: {
+                        filename: record['encoded_id'],
+                        'store_id': config.storeId,
+                        'as_is': 1,
+                        'force_static_path': targetElement.data('force_static_path') ? 1 : 0,
+                        'form_key': FORM_KEY
+                    },
+                    context: this,
+                    showLoader: true
+                }).done($.proxy(function (data) {
+                    $.mage.mediabrowser().insertAtCursor(targetElement.get(0), data);
+                }, this));
+            } else {
+                targetElement.val(record['thumbnail_url'])
+                    .data('size', record.size)
+                    .data('mime-type', record['content_type'])
+                    .trigger('change');
+            }
+            window.MediabrowserUtility.closeDialog();
+            targetElement.focus();
+        },
+
+        /**
+         * Return opener Window object if it exists, not closed and editor is active
+         *
+         * @param {String} targetElementId
+         * return {Object|null}
+         */
+        getMediaBrowserOpener: function (targetElementId) {
+            if (typeof wysiwyg != 'undefined' &&
```

Comment:
Can you please explain why we check is Wysiwyg and tinyMceEditors exists? If I correct to understand, wysiwyg can be except in the case when it is disabled in the configuration, correct? in this case, we should pass is wysiwyg enabled option by the configuration, not just check is JS class exists. It's not very important for now, but we need at least should create task for that @sivaschenko

Reply:
thank you, i agree with, for now i added a proper check with isUndefined, and later i might refactor it to use configuration settings.